edsm.py

```diff
@@ -102,23 +159,46 @@ def worker(self, system_name, result):
 
 
 # Flight log - http://www.edsm.net/api-logs
-def export(data, edsmlookupfn):
-
+def export(data, coordinates, edsmlookupfn):
     querytime = config.getint('querytime') or int(time.time())
-
-    writelog(querytime, data['lastSystem']['name'], edsmlookupfn)
+    writelog(querytime, data['lastSystem']['name'], coordinates, edsmlookupfn)
 
 
-def writelog(timestamp, system, edsmlookupfn):
+def writelog(timestamp, system, coordinates, edsmlookupfn):
 
     try:
         # Look up the system before adding it to the log, since adding it to the log has the side-effect of creating it
         edsmlookupfn()
 
         if system in EDSM.FAKE:
             return
-
-        r = requests.get('http://www.edsm.net/api-logs-v1/set-log?commanderName=%s&apiKey=%s&systemName=%s&dateVisited=%s&fromSoftware=%s&fromSoftwareVersion=%s' % (urllib.quote(config.get('edsm_cmdrname').encode('utf-8')), urllib.quote(config.get('edsm_apikey')), urllib.quote(system), urllib.quote(time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(timestamp))), urllib.quote(applongname), urllib.quote(appversion)), timeout=EDSM._TIMEOUT)
+            
+        if coordinates:
+            (x,y,z) = coordinates
+            url = '%s/api-logs-v1/set-log?commanderName=%s&apiKey=%s&systemName=%s&dateVisited=%s&x=%s&y=%s&z=%s&fromSoftware=%s&fromSoftwareVersion=%s' % (
+                self.endPoint,
```

Comment:
The code fails to lookup the endPoint value here since the writelog method is not part of the EDSM class above.  I had to move the endPoint field in the EDSM class above to a global variable in this file to use it in this method.  Other than that it seems to be working.